autotest-agent/src/jvmMain/kotlin/com/epam/drill/test/agent/TestListener.kt

```diff
@@ -104,44 +75,30 @@ object TestListener {
         methodParams: String = "()",
         classParams: String = "",
     ) {
-        if (className != null && method != null) {
-            val test = TestDetails(
-                engine = engine,
-                path = className,
-                testName = method,
-                params = mapOf(
-                    classParamsKey to classParams,
-                    methodParamsKey to methodParams,
-                ),
-            )
-            logger.trace { "Test: $test is finishing with status $status..." }
-            if (isNotFinalizeTestState(test)) {
-                addTestResult(test, status)
-            }
-        }
-    }
-
-    private fun addTestResult(test: TestDetails?, status: String) {
-        test?.takeIf { it in _testInfo.value }?.let {
-            addTestInfo(
-                test,
-                TestInfo::finishedAt.name to System.currentTimeMillis(),
-                TestInfo::result.name to getByMapping(status)
-            )
-            logger.debug { "Test: $test FINISHED. Result:$status" }
-            clearDrillHeaders(it)
-            if (Configuration.parameters[ParameterDefinitions.WITH_JS_COVERAGE]) sendSessionData(test.hash())
+        if (className == null || method == null)
```

Comment:
This was the code before my changes, I just reduced the nesting. I guess it might have been due to the difficulty of finding the ideal point to call this function in the test frameworks.